src/test/unit/io/reader_test.cpp

```diff
@@ -1429,3 +1429,93 @@ TEST(IoReader, UnitVectorThrows) {
   EXPECT_THROW(reader.unit_vector_constrain(x), std::invalid_argument);
   EXPECT_THROW(reader.unit_vector_constrain(x, lp), std::invalid_argument);
 }
+
+TEST(IoReader, var_vector) {
```

Comment:
Since there's the `to_var_value` here it's probably worth adding this to all the var tests:

```
vec_x.adj() = Eigen::VectorXd::Random(vec_x.size());
stan::math::grad();
for(size_t i = 0;  i < vec_x.size(); ++i) {
  EXPECT_FLOAT_EQ(vec_x.adj()(i), theta[i].adj());
}
```

I know the other tests don't do it -- just looks like something that would be easy to miss if someone comes through in the future and easy to test.

Reply:
This we talked about and I'm fine not having a test